pkg/dnscache/dnscache.go

```diff
@@ -0,0 +1,204 @@
+package dnscache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/golang-lru/v2/expirable"
+
+	"github.com/aquasecurity/tracee/pkg/utils/set"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+var (
+	ErrDnsCacheIsNotInitialized = errors.New("dns cache is not initialized")
+)
+
+const (
+	dnsCacheSize        = 500
+	dnsCacheTTL         = time.Minute * 5
```

Comment:
For TTL, we have to distinguish the 2 different TTL concepts (at least internally) I believe.

1) The name resolution TTL: used by the resolvers (clients) mostly.
2) A internal cache timeout of a cached entry.

Some people say it should be the same as the resolution TTL.

A problem that I thought (for you to discuss with the sponsors of this need being worked):

When doing round-robin load balancing, for example, some DNS servers might return ALL IPs for a specific A, CNAME, AAAA query. Then the client decides which one to use. Some other DNS servers might return one different IP each time you resolve that same A, CNAME, AAAA entry, for example.

My point is, if this datasource is ever used to distinguish to which host a process talked to, then we will need to correlate the process and a specific entry in this `Set` you created (and not index by name used, as one process might have talked to IP1, the other to IP2, the other to IP3, and all of them talking to the same name, and vice-versa).

Reply:
So about the two TTL types, since this is based on what the sponsors originally developed, I assume they are ok with only using an internal cache TTL. We could, in addition, develop an Expirable generic struct and make the set items of this type, WDYT? 

About the process distinction:
If this is something we want I believe we have two options I can think of:
1. Make the set entry a struct which includes the process which communicated to it (like storing a "foreign key" to the process tree
2. The process tree could hold a dns cache per process (doesn't sound right to me)